pkg/util/ciutil/detect.go

```diff
@@ -74,7 +74,7 @@ var detectors = map[SystemName]system{
 	GenericCI: genericCICI{
 		baseCI: baseCI{
 			Name:            SystemName(os.Getenv("PULUMI_CI_SYSTEM")),
-			EnvVarsToDetect: []string{"GENERIC_CI_SYSTEM"},
+			EnvVarsToDetect: []string{"PULUMI_CI_SYSTEM"},
```

Comment:
@chrsmith From our last discussion about this, we called the CI System as Generic CI on-purpose. Please see the comment on line #70 (above the GenericCI map entry). I believe only line 76 needs to be corrected to use `GENERIC_CI_SYSTEM`. I think that's the bug here. Personally calling this map entry a GenericCI and then requiring the env vars to be PULUMI_* doesn't make sense to me. Let me know what you think.

Reply:
I'm a little confused by what you mean. But let me answer two variants of what I think you are saying:

---

It seems unnecessary to have the user specify two different environment variables to indicate they are on our "generic CI" system / "unsupported CI escape hatch".

Why set _both_ `GENERIC_CI_System` AND `PULUMI_CI_SYSTEM` when you could just set one? (Which has a nice side effect of requiring you to provide the name of the CI system that is unsupported, etc.)

---

The thing I'd like to address here is to make sure that with our generic system the user can pass along the name of their CI system. For example, when running on GitHub Actions we have:

`export PULUMI_CI_SYSTEM="GitHub"`
https://github.com/pulumi/pulumi/blob/master/dist/actions/entrypoint.sh#L18

It seems like it it would be a shame if the user did provide all the right metadata, but we just attributed it to "Generic CI System". Though for "belt and suspenders" in this PR I also explicit set the system name again in `generic.go`:

```
v.Name = SystemName(os.Getenv("PULUMI_CI_SYSTEM"))
```

---

Does that make sense? Or am I missing something?

Reply:
Yeah we are likely saying the same thing.

> Why set both GENERIC_CI_System AND PULUMI_CI_SYSTEM when you could just set one? (Which has a nice side effect of requiring you to provide the name of the CI system that is unsupported, etc.)

I do agree with this and that's why I said this: `I believe only line 76 needs to be corrected to use GENERIC_CI_SYSTEM. I think that's the bug here.`

> It seems like it it would be a shame if the user did provide all the right metadata, but we just attributed it to "Generic CI System".

You are right and I am not suggesting that we attribute it to a "Generic CI System". The CI system name would come from line 76. So whatever env var that is looking at for the name is all that we need to correct.

> Though for "belt and suspenders" in this PR I also explicit set the system name again in generic.go:
> v.Name = SystemName(os.Getenv("PULUMI_CI_SYSTEM"))

I don't think this is necessary separately. This is why in `detect.go` we set the `Name` property to be the value of an env var that would have the name of the unknown (to Pulumi) CI system name.